client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -541,4 +541,13 @@ public void close() {
         synchronizer.close();
         sequentialProcessor.close();
     }
+
+    /**
+     * Cancels the outStanding checkpoints.
+     * @returns true if the outStanding checkpoints are cleared.
+     */
+    public boolean cancelOutstandingCheckpoints() {
+        ReaderGroupState state = synchronizer.getState();
+        return state.getCheckpointState().removeOutstandingCheckpoints();
```

Comment:
Do not modify the state synchronizer state without going through the synchronizer. This will result in different hosts having different states. There needs to be an update to the ReaderGroupState object to cancel the checkpoints. See the stateSynchronizer documentation and the existing updates in ReaderGroupState to see how this is done.

Reply:
I have implemented the above suggestion.